readbuffer.go

```diff
@@ -64,6 +72,11 @@ func (r *readBuffer) Read(b []byte) (int, error) {
 			n, err = r.buf.Read(b)
 			r.cond.Broadcast()
 			if err != io.EOF {
+				if r.buf.Len() < MaxBuffer/8 {
```

Comment:
Is there a chance when the buf always has more data than MaxBuffer/8 so that it never resume other write go routines?

Reply:
I believe this question remains. It is possible that on one read, the `err == nil` and `r.buf.Len() >= MaxBuffer/8`? Then on the next read `err == io.EOF`?

It would seem that in this case we would never resume.

Reply:
The above comment is not relevant. The implementation of `bytes.Reader` will return `io.EOF` only if the buffer is empty on read. If the buffer is empty after the read, then a `nil` error is returned. So this looks correct to me.